cmake/app/boilerplate.cmake

```diff
@@ -558,6 +558,14 @@ set(SOC_SERIES ${CONFIG_SOC_SERIES})
 set(SOC_TOOLCHAIN_NAME ${CONFIG_SOC_TOOLCHAIN_NAME})
 set(SOC_FAMILY ${CONFIG_SOC_FAMILY})
 
+# For the gen_app_partitions.py to work correctly, we must ensure that
+# all targets exports their compile commands to fetch object files.
+# We enable it unconditionally, as this is also useful for several IDEs
+set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE CACHE BOOL
+    "Export CMake compile commands. Used by gen_app_partitions.py script"
+    FORCE
+)
+
```

Comment:
Checked the overhead for this.

Was unable to measure a time overhead, and the file generated was 244k.

So generating this unconditionally is OK.

Reply:
Looks like we can now get rid of some code in twister.

/home/sebo/ncs/zephyr/scripts/pylib/twister/twisterlib.py
2045:            cmake_args.append("-DCMAKE_EXPORT_COMPILE_COMMANDS=1")

Reply:
> Was unable to measure a time overhead, and the file generated was 244k.
>
> So generating this unconditionally is OK.

Thanks for checking.
Initial version was actually doing this conditionally, but feedback on Discord suggested to do this unconditionally.

So great you checked the impact.

Reply:
Done, `CMAKE_EXPORT_COMPILE_COMMANDS` removed in twister.